doc/generate2.cpp

```diff
@@ -29,7 +31,16 @@ int main(int argc, char *argv[])
 				std::cout <<  "'" << cmd << "' ";
 				std::cout << "(default key: '" << key << "')::\n";
 				std::cout << "         " << desc << "\n\n";
+			} else {
+				// should never happen
+				std::cerr << "expected exactly 3 cells in " << argv[1] << ":" << lineno;
```

Comment:
This text made me realize that the regular expression above is wrong: it matches three *or more* cells, because the last part is `(.*)`, not `([^:}*)`. Can you make it so? Then we can get rid of the whole `matches.size() == 4` check (return code of `regex_match` is sufficient), though the error message will become more vague (because we can no longer say how many cells we found—only that there wasn't three of them).

Reply:
Assuming your closing curly brace should be a closing square bracket `([^:]*)`, this won't work, because then the third column cannot contain escaped colons. However this is the case in line 28 of _doc/keycmds.dsv_.

I rather suggest to either leave the regex like this or else switch to double pipes for this DSV file, too. What do you reckon?

Reply:
Assumption is correct, sorry about the typo. Switching to double pipes sounds like a better solution to me because it lets us make `generate2` do more checks, which is good.

Reply:
Alright, let's use double pipes.